apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+  },
+  {
+    owner_evaluations: [{ review_decision: 'reviewed' as EnumPullReqReviewDecision }]
+  }
+]
+// Mock data for extractInfoFromRuleViolationArr
+const mockRuleViolations = [
+  {
+    violations: [{ message: 'Violation 1' }, { message: 'Violation 2' }]
+  },
+  {
+    violations: [{ message: 'Violation 3' }]
+  }
+]
+
+// Tests
+describe('processReviewDecision', () => {
+  it('should mark review as outdated if SHAs differ', () => {
+    const result = processReviewDecision(mockReviewDecision, mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(PullReqReviewDecision.outdated)
+  })
+})
+
+describe('generateStatusSummary', () => {
+  it('should generate a summary of status checks', () => {
+    const result = generateStatusSummary(mockChecks)
+    expect(result.summary.total).toBe(2)
+    expect(result.summary.successReq).toBe(1)
+    expect(result.summary.failed).toBe(1)
+  })
+})
+
+// Mock data for determineStatusMessage
+const mockChecksForStatusMessage = [
+  { check: { status: 'success' as EnumCheckStatus }, required: true },
+  { check: { status: 'failure' as EnumCheckStatus }, required: false }
+]
+
+describe('determineStatusMessage', () => {
+  it('should determine the status message based on the status summary', () => {
+    const result = determineStatusMessage(mockChecksForStatusMessage as TypeCheckData[])
+    expect(result).toEqual({
+      color: 'text-success',
+      content: '1 failed, 1 succeeded',
+      status: 'success',
+      title: 'All required checks passed'
+    })
+  })
+})
+
+describe('extractSpecificViolations', () => {
+  it('should extract specific violations based on rule', () => {
+    const result = extractSpecificViolations(mockViolationsData, 'rule1')
+    expect(result).toEqual([{ code: 'rule1', description: 'Violation 1' }])
+  })
+})
+
+describe('checkIfOutdatedSha', () => {
+  it('should return true if SHAs differ', () => {
+    const result = checkIfOutdatedSha(mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(true)
+  })
+})
+
+describe('capitalizeFirstLetter', () => {
+  it('should capitalize the first letter of a string', () => {
+    const result = capitalizeFirstLetter('hello')
+    expect(result).toBe('Hello')
+  })
+})
+
+describe('findChangeReqDecisions', () => {
+  it('should find change request decisions from entries', () => {
+    const result = findChangeReqDecisions(mockEntriesForChangeReq, 'approve')
+    expect(result).toEqual([
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      },
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      }
+    ])
+  })
+})
+
+describe('findWaitingDecisions', () => {
+  it('should find waiting decisions from entries', () => {
+    const result = findWaitingDecisions(mockEntriesForWaitingDecisions)
+    expect(result).toEqual([])
+  })
+})
+
+describe('normalizeGitFilePath', () => {
+  it('should normalize file paths with spaces', () => {
```

Comment:
This seems to prove that nothing will happen if the path as a space. Can we get a test that prove what this function will do?

Reply:
The function checks if the file path ends with a tab character (\t) and contains a space. If both conditions are met, it removes the trailing tab character. added a test to check that

Reply:
initially understood it wrong